File: backend/src/main/java/com/isravel/workhub/payroll/PayrollRecord.java
package com.isravel.workhub.payroll;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payroll_records")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PayrollRecord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_id", nullable = false)
    private Long employeeId;

    @Column(name = "month", nullable = false)
    private Integer month;

    @Column(name = "year", nullable = false)
    private Integer year;

    @Column(name = "regular_hours")
    private BigDecimal regularHours = BigDecimal.ZERO;

    @Column(name = "overtime_hours")
    private BigDecimal overtimeHours = BigDecimal.ZERO;

    @Column(name = "hourly_rate", nullable = false)
    private BigDecimal hourlyRate;

    @Column(name = "overtime_multiplier")
    private BigDecimal overtimeMultiplier = new BigDecimal("1.50");

    @Column(name = "gross_pay")
    private BigDecimal grossPay = BigDecimal.ZERO;

    @Column(name = "deductions")
    private BigDecimal deductions = BigDecimal.ZERO;

    @Column(name = "bonuses")
    private BigDecimal bonuses = BigDecimal.ZERO;

    @Column(name = "net_pay")
    private BigDecimal netPay = BigDecimal.ZERO;

    @Column(name = "status")
    private String status = "PENDING";

    @Column(name = "processed_date")
    private LocalDateTime processedDate;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: flutter_app/lib/models/payroll_record.dart
class PayrollRecord {
  final int id;
  final int employeeId;
  final int? payPeriodId;
  final int month;
  final int year;
  final double regularHours;
  final double overtimeHours;
  final double paidHours;
  final double breakHours;
  final double hourlyRate;
  final double overtimeMultiplier;
  final double grossPay;
  final double deductions;
  final double bonuses;
  final double netPay;
  final String status;
  final String? processedDate;

  PayrollRecord({
    required this.id,
    required this.employeeId,
    this.payPeriodId,
    required this.month,
    required this.year,
    required this.regularHours,
    required this.overtimeHours,
    required this.paidHours,
    required this.breakHours,
    required this.hourlyRate,
    required this.overtimeMultiplier,
    required this.grossPay,
    required this.deductions,
    required this.bonuses,
    required this.netPay,
    required this.status,
    this.processedDate,
  });

  factory PayrollRecord.fromJson(Map<String, dynamic> json) {
    return PayrollRecord(
      id: json['id'],
      employeeId: json['employeeId'],
      payPeriodId: json['payPeriodId'],
      month: json['month'],
      year: json['year'],
      regularHours: (json['regularHours'] as num).toDouble(),
      overtimeHours: (json['overtimeHours'] as num).toDouble(),
      paidHours: (json['paidHours'] as num? ?? json['regularHours'] as num).toDouble(),
      breakHours: (json['breakHours'] as num? ?? 0).toDouble(),
      hourlyRate: (json['hourlyRate'] as num).toDouble(),
      overtimeMultiplier: (json['overtimeMultiplier'] as num).toDouble(),
      grossPay: (json['grossPay'] as num).toDouble(),
      deductions: (json['deductions'] as num).toDouble(),
      bonuses: (json['bonuses'] as num).toDouble(),
      netPay: (json['netPay'] as num).toDouble(),
      status: json['status'] ?? 'PENDING',
      processedDate: json['processedDate'],
    );
  }
}

